File: front-end/src/app/service-models/app/v1/client/client-employee-additional-reimbursement-log-s-m.ts
import { CoinManagementServiceModelBase } from '../../base/coin-management-service-model-base';
import { ReimbursementTypeSM } from '../../enums/reimbursement-type-s-m.enum';

export class ClientEmployeeAdditionalReimbursementLogSM extends CoinManagementServiceModelBase<number> {
    clientUserId?: number;
    reimbursementType!: ReimbursementTypeSM;
    reimburseDocumentName!: string;
    extension!: string;
    reimbursementDescription!: string;
    reimbursementAmount!: number;
    reimbursementDate!: Date;
    reimbursementDocumentPath!: string;
    clientCompanyDetailId!: number;
}

File: backend/payroll-backend/Components/Siffrum.Web.Payroll.ServiceModels/v1/Client/ClientEmployeeAdditionalReimbursementLogSM.cs
using Siffrum.Web.Payroll.ServiceModels.Base;
using Siffrum.Web.Payroll.ServiceModels.Enums;

namespace Siffrum.Web.Payroll.ServiceModels.v1.Client
{
    public class ClientEmployeeAdditionalReimbursementLogSM : SiffrumPayrollServiceModelBase<int>
    {
        public int? ClientUserId { get; set; }
        public ReimbursementTypeSM ReimbursementType { get; set; }
        public string ReimburseDocumentName { get; set; }
        public string Extension { get; set; }
        public string ReimbursementDescription { get; set; }
        public decimal ReimbursementAmount { get; set; }
        public DateTime ReimbursementDate { get; set; }
        public string ReimbursementDocumentPath { get; set; }
        public int ClientCompanyDetailId { get; set; }
    }
}
